### PR TITLE
[fix] Fix the prepare debug scripts

### PR DESCRIPTION
**Pull request by steakhal**
There are multiple changes in this commit:
- Expect 'compilation_database.json' instead of 'compile_cmd.json' in
  failed zips.
- Uplift the 'compiler_info.json' preparation to the 'new' format.
- Use the expected compiler for '--ctu-collect'.

Since these were broken for quite a while, and these debug scripts are
not for production use, I'm not willing to create tests for this change.

### CODE REVIEW

File: scripts/debug_tools/prepare_all_cmd_for_ctu.py [hunk 1]
```diff
@@ -79,13 +80,13 @@ def get_triple_arch(analyze_command_file):
...
 
-    compile_cmd_debug = "compile_cmd_DEBUG.json"
+    compile_cmd_debug = "compilation_database_DEBUG.json"
     with open(compile_cmd_debug, 'w',
               encoding="utf-8", errors="ignore") as f:
         f.write(
             json.dumps(
                 prepare_compile_cmd.prepare(
-                    os.path.join(args.report_dir, "compile_cmd.json"),
+                    os.path.join(args.report_dir, "compilation_database.json"),
```
Comment (csordasmarton) [inline scripts/debug_tools/prepare_all_cmd_for_ctu.py:89]: Here you expect `compilation_database.json` in the report directory but CodeChecker will create a `compile_cmd.json` to the report directory when you run the `CodeChecker analyze` command.

See:
https://github.com/Ericsson/codechecker/blob/65f8a5056ee6e86661ae799708f8ec4392a5e67c/analyzer/codechecker_analyzer/cmd/analyze.py#L1037

Comment (steakhal) [reply scripts/debug_tools/prepare_all_cmd_for_ctu.py:89]: Well, then we should consider updating that behavior to prefer using `compilation_database.json`

Comment (steakhal) [reply scripts/debug_tools/prepare_all_cmd_for_ctu.py:98]: I decided to support the `compile_cmd.json` via moving it to `compilation_database.json`.
It's basically canonicalizing to the latter.

**Review by csordasmarton — APPROVED** [review]
LGTM!

**Pull request merged**